js/src/UXCapture.js

```diff
@@ -2,44 +2,51 @@ import ExpectedMark from "./ExpectedMark";
 import View from "./View";
 
 const NOOP = () => {};
+const INTERACTION_START_MARK = "interactionStart";
+const NAVIGATION_START_MARK = "navigationStart";
 
 export default class UXCapture {
   constructor() {
     this.onMark = NOOP;
     this.onMeasure = NOOP;
+    this.startMark = NAVIGATION_START_MARK;
 
     window.UX = this;
```

Comment:
Also a note that I think @sergeychernyshev was going to remove this assignment and leave it to the caller? I don't think there's a need for the library to attach itself to `window`.

Reply:
No, we do need it so SSR code can call UX.mark() on the page.

I already removed keeping window on UX object as we just use one global window everywhere now.

Reply:
we can call `UX.mark` in the SSR code and still give the responsibility for assigning `window.UX` to the consumer app as long as `UX.mark` is called after the `UXCapture` instance is initialized. 

We would start by making the 'export' code in `ux-capture.js` more consistent, e.g. [this line](https://github.com/sergeychernyshev/ux-capture/blob/94cf45a504f46655b77a5a928b44d73b22b7ae3e/js/src/ux-capture.js#L18) would be

```js
window.UXCapture = UXCapture;
```

which makes it consistent with the `module.exports = UXCapture` case and the `define([], function() { return UXCapture; });` case, neither of which instantiate the object. The current interface is "if you execute this directly in the browser, you don't have to instantiate `UXCapture`, but if you use AMD, then you do", which is not an intrinsically necessary distinction.

If we exported the `UXCapture` class instead of the `window.UX` instance, instead of calling `window.UX.config`, the client would instantiate `window.UX` itself [in UXCaptureConfig](https://github.com/meetup/meetup-web-platform/blob/master/packages/mwp-app-render/src/components/uxcapture/UXCaptureConfig.jsx#L19) with something like

```jsx
<script>
  if(window.UXCapture) {
    window.UX = new window.UXCapture({
      ${onMark ? `"onMark": ${onMark.toString()},` : ''}
      ${onMeasure ? `"onMeasure": ${onMeasure.toString()},` : ''}
    });
  }
</script>
```

That approach has a few benefits in terms of making the usage/implementation more clear.

1. The 'export' cases in ux-capture.js are consistent - initializing `UXCapture` is _always_ the responsibility of the consumer
2. The consumer can decide how it wants to name the UXCapture instance - we define the class interface, they define how instances are managed
3. no need for a lazy `.config` method with unclear rules about when it can/should be called - the `UXCapture` instance is configured once on initialization by definition

Reply:
Let's keep track of this thread, but break it out into a separate conversation. I like aspects of this approach, but I think it's beyond the scope of this ticket. Perhaps we can file a separate ticket for it, and you both can pair on it?

Reply:
definitely out of scope - I can write up a ticket